tests/ci.js

```diff
@@ -1,181 +1,204 @@
-  }
+function getDetailedMessage(message, runResult) {
+  return message + '\n\n' + 'STDOUT\n' + runResult.stdout + '\n\n' + 'STDERR\n' + runResult.stderr;
 }
 
-function assertTestSuccess(testFile, clearCache) {
-  const result = run(testFile, clearCache);
-  if (result.code !== 0) {
-    fail('Expected tests to pass with exit code 0.', result);
-  }
+function assertTestSuccess(runResult) {
+  const msg = 'Expected success (exit code ' + resultSuccess + '), but got ' + runResult.status;
+  assert.strictEqual(resultSuccess, runResult.status, getDetailedMessage(msg, runResult));  
 }
 
-function fail(message, result) {
-  shell.echo();
-  shell.echo(`######### ERROR`);
-  shell.echo(
-    "The last elm-test run above didn't run as expected. Details below."
-  );
-  shell.echo();
-  shell.echo('### stdout');
-  shell.echo(result.stdout || '(no stdout)');
-  shell.echo();
-  shell.echo('### stderr');
-  shell.echo(result.stderr || '(no stderr)');
-  shell.echo();
-  shell.echo('### message');
-  shell.echo(message);
-  shell.echo(`Exit code: ${result.code}. See stdout and stderr above.`);
-  shell.echo(`This message comes from tests/${filename}`);
-  shell.echo();
-  shell.exit(1);
+function assertTestErrored(runResult) {
+  const msg = 'Expected error (exit code ' + resultErrored + '), but got ' + runResult.status;
+  assert.strictEqual(resultErrored, runResult.status, getDetailedMessage(msg, runResult));  
 }
 
-shell.echo(filename + ': Uninstalling old elm-test...');
-shell.exec('npm remove --ignore-scripts=false --global ' + elmTest);
-
-shell.echo(filename + ': Installing elm-test...');
-shell.exec('npm link --ignore-scripts=false');
-
-var interfacePath = require('elmi-to-json').paths['elmi-to-json'];
-
-shell.echo(filename + ': Verifying elmi-to-json is installed...');
-var interfaceResult = spawn.sync(interfacePath, ['--help']);
-var interfaceExitCode = interfaceResult.status;
-
-if (interfaceExitCode !== 0) {
-  shell.echo(
-    filename +
-      ': Failed because `elmi-to-json` is present, but `elmi-to-json --help` returned with exit code ' +
-      interfaceExitCode
-  );
-  shell.echo(interfaceResult.stdout.toString());
-  shell.echo(interfaceResult.stderr.toString());
-  shell.exit(1);
+function assertTestFailure(runResult) {
+  const msg = 'Expected failure (exit code >= ' + resultFailureThreshold + '), but got ' + runResult.status;
+  assert.ok(runResult.status >= resultFailureThreshold, getDetailedMessage(msg, runResult));  
 }
 
-shell.exec('npm link --ignore-scripts=false');
-
-shell.echo(filename + ': Verifying installed elm-test version...');
-var versionRun = shell.exec(elmTest + ' --version');
-
-if (versionRun.code !== 0) {
-  shell.exec(
-    'echo Expected elm-test --version to exit with exit code 0, but it was ' +
-      versionRun.code
-  );
-  shell.exit(1);
-}
+describe('--help', () => {
+  it('Should exit indicating success', () => {
+    const runResult = execElmiToJSON(['--help']);
```

Comment:
To be completely honest, I wasn't 100% sure about this myself. Here's the snippet from the original:
```
shell.echo(filename + ': Verifying elmi-to-json is installed...');
var interfaceResult = spawn.sync(interfacePath, ['--help']);
var interfaceExitCode = interfaceResult.status;

if (interfaceExitCode !== 0) {
  shell.echo(
    filename +
      ': Failed because `elmi-to-json` is present, but `elmi-to-json --help` returned with exit code ' +
      interfaceExitCode
...
```
As far as I understand it, this simply verifies that elmi-to-json is present. Can we get rid of the elmi-to-json part altogether?

Reply:
I think so, yes! Because:

- I don’t understand why it’s validated in the first place
- I’m removing elmi-to-json altogether in #442

Reply:
I've removed everything related to elmi-to-json.